hangman.py

```diff
@@ -0,0 +1,308 @@
+    '''
+    secret_word: string, the word the user is guessing; assumes all letters are
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    if secret_word == ''.join(letters_guessed):
+        return True
+    else:
+        return False
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    t_word = ""
+    for i in secret_word:
+        if i in letters_guessed:
+            t_word += i
+        else:
+            t_word += ("_ ")
+    return (t_word)
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    ran = string.ascii_lowercase
+    available_letters = ""
+    for i in range(len(ran)):
+        if ran[i] not in letters_guessed:
+            available_letters += ran[i]
+    return (available_letters)
+
+
+def hangman(secret_word):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+
+    guesses_remaining = 6
+    warnings_remaining = 3
+    letters_guessed = []
+    print('''Welcome to the game Hangman!
+I am thinking of a word that is {} letters long.
+You have {} warnings left'''.format(len(secret_word), warnings_remaining))
+    while (guesses_remaining > 0):
+        print('''-------------
+You have {} guesses left.
+Available letters: {}'''.format(guesses_remaining, get_available_letters(letters_guessed)))
+
+        letters_guessed.append((input("Please guess a letter:")).lower())
+
+        if secret_word == get_guessed_word(secret_word, letters_guessed):
```

Comment:
Actually you have a separate function for that, called `is_word_guessed` - try to use it